source/MRMesh/MRCone3.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+
+// Base class for cone parameterization
+
+template <typename T>
+class Cone3
+{
+public:
+    Cone3()
+    {}
+
+    Cone3( const Line3<T>& inAxis, T inAngle, T inLength )
+        :
+        position( inAxis ),
+        angle( inAngle ),
+        length( inLength )
+    {}
+
+    inline MR::Vector3<T>& center( void )
+    {
+        return position.p;
+    }
+    inline MR::Vector3<T>& direction( void )
+    {
+        return position.d;
+    }
+    inline MR::Vector3<T>& apex( void )
+    {
+        return center();
+    }
```

Comment:
why apex is same as center?

Reply:
It seems to me reasonable to have a center method for all features, so that it is convenient to measure the distances to the centers of any feature with any other in the future.